### PR TITLE
[RLlib] PyTorch version of ES (Evolution Strategies).

### PR DESCRIPTION
**Pull request by sven1977**
This PR implements a PyTorch version of RLlib's ES algorithm using RLlib's functional algo builder API. It also adds a regression test for ES (torch) on CartPole.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by ericl — APPROVED** [review]
Looks good. Can we benchmark Humanoid before / after / TF / Torch on a single big machine?

File: rllib/agents/es/es_torch_policy.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        for k in sorted(theta_dict.keys()):
+            shape = policy.param_shapes[k]
+            num_params = int(np.prod(shape))
+            new_theta_dict[k] = torch.from_numpy(
+                np.reshape(theta[pos:pos + num_params], shape))
+            pos += num_params
+        policy.model.load_state_dict(new_theta_dict)
+
+    def _get_flat_weights(policy):
```
Comment (ericl) [inline rllib/agents/es/es_torch_policy.py:41]: Should these be common methods of TorchPolicy?

File: doc/source/rllib-algorithms.rst [hunk 2]
```diff
@@ -422,7 +422,7 @@ Tuned examples: `CartPole-v0 <https://github.com/ray-project/ray/blob/master/rll
 
 Evolution Strategies
 --------------------
-|tensorflow|
+|pytorch| |tensorflow|
```
Comment (ericl) [inline doc/source/rllib-algorithms.rst:425]: Don't forget to update the sigils in rllib-toc.rst (should also add for DQN, etc. missed in the past)

Comment (sven1977) [reply doc/source/rllib-algorithms.rst:425]: +1 Will do.

**Comment by sven1977** [comment]
Thanks! Yes, will add a nice benchmark (tf vs torch) to rl-experiments for ES (we don't have any right now). On my sprint this week :)

File: rllib/agents/es/es_torch_policy.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/es/es_torch_policy.py:41]: Not sure, but it's high on my TODO list to unify these with tf/eager policies: Return a unified dict with identical keys/equally sorted list for two Policies that have equivalent Models.

**Pull request merged**